Resources/doc/reference/security.rst

```diff
@@ -74,89 +74,97 @@ Let's create the following strategy : a media can be downloaded only once per se
 
     use Sonata\MediaBundle\Model\MediaInterface;
     use Symfony\Component\HttpFoundation\Request;
-    use Symfony\Component\Security\Core\SecurityContextInterface;
+    use Symfony\Component\HttpFoundation\Session\Session;
     use Symfony\Component\Translation\TranslatorInterface;
-    use Symfony\Component\DependencyInjection\ContainerInterface;
 
     class SessionDownloadStrategy implements DownloadStrategyInterface
     {
-        protected $container;
+        /**
+         * @var Session
+         */
+        protected $session;
 
+        /**
+         * @var TranslatorInterface
+         */
         protected $translator;
 
+        /**
+         * var int
+         */
         protected $times;
 
+        /**
+         * @var string
+         */
         protected $sessionKey = 'sonata/media/session/times';
 
         /**
-         * @param \Symfony\Component\Translation\TranslatorInterface $translator
-         * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
-         * @param int $times
+         * @param TranslatorInterface $translator
+         * @param Session             $session
+         * @param int                 $times
          */
-        public function __construct(TranslatorInterface $translator, ContainerInterface $container, $times)
+        public function __construct(TranslatorInterface $translator, Session $session, $times)
```

Comment:
Instead of using the concrete implementation, please use the Interface `Symfony\Component\HttpFoundation\Session\SessionInterface` .